src/ledger/LedgerTxnEntry.cpp

```diff
@@ -55,12 +55,13 @@ toEntryImplBase(std::shared_ptr<LedgerTxnEntry::Impl> const& impl)
     return impl;
 }
 
-LedgerTxnEntry::LedgerTxnEntry()
+LedgerTxnEntry::LedgerTxnEntry() : mIsInitialized(false)
 {
 }
 
 LedgerTxnEntry::LedgerTxnEntry(std::shared_ptr<Impl> const& impl)
     : mImpl(impl) // Constructing weak_ptr from shared_ptr is noexcept
+    , mIsInitialized(true)
```

Comment:
I don't think this is wrong today, but it is a foot-gun. I think I would be surprised if `LedgerTxnEntry()` is not identical to `LedgerTxnEntry(nullptr)`. We should probably guard against this possibility, perhaps by making them identical via constructor delegation. I'm open to other suggestions here.

Reply:
Updated to assign `mIsInitialized` using the `shared_ptr` bool operator.